plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -788,6 +790,14 @@ def get_table_options(
             labels = config.get('labels', {})
             opts['labels'] = list(labels.items())
 
+        if config.get('require_partition_filter'):
+            opts['require_partition_filter'] = config.get(
+                'require_partition_filter')
+
+        if config.get('partition_expiration_days') is not None:
+            opts['partition_expiration_days'] = config.get(
+                'partition_expiration_days')
```

Comment:
I support naming this `partition_expiration_days`, even though it sounds a little clunky, because it exactly matches what BigQuery calls this.

I'm now thinking that, for clarity, we might rename `hours_to_expiration` to `table_expiration_hours`. (With backwards compatibility by still checking `hours_to_expiration`, of course)

Reply:
Sounds good. In my opinion, using the same names is easy for BigQuery users to configure dbt models.

Reply:
Will implement.

> Regardless, it'd be good to add an integration test to test_bigquery_changing_partitions where we switch from:
>
> * a table with None for partition_expiration_days and require_partition_filter
> * a table with values for both